src/mrfgen/mrfgen.py

```diff
@@ -536,13 +546,108 @@ def crop_to_extents(tile, tile_extents, projection_extents, working_dir):
+        self.cond.release()
+
+lock = rw_lock() # used to ensure that gdal_merge doesn't happen at the same time as a parallel insert
+
+def parallel_mrf_insert(tiles, mrf, insert_method, resize_resampling, target_x, target_y, mrf_blocksize,
+                   target_extents, target_epsg, nodata, merge, working_dir, no_cpus):
+    """
+    Launches multiple workers each handling a fraction of the tiles to be merged into the final mrf file. 
+    Also sets the mrf to be mp_safe to allow for simultaneous access. Generally 2-6 workers is ideal. 
+    There can be some degredation in performance for large numbers of workers. We use the rw_lock to 
+    synchronize access between the processes, since gdal_merge must be performed while no other process is running
+    mrf_insert.
+
+    Arguments:
+        tiles ... working_dir: Same as mrf_insert
+        no_cpus (int) -- Number of CPUs to run mrf_insert in parallel
+    """
+
+    log_info_mssg("parallel_mrf_insert with mrf {}".format(mrf))
+
+    no_pools = min(multiprocessing.cpu_count() - 1, len(tiles), no_cpus)
+    log_info_mssg("no_pools for parallel mrf_insert is {} for mrf {}".format(no_pools, mrf))
+
+    if len(tiles) == 1 or no_pools == 1:
+        log_info_mssg("making serial call since not enough tiles or cores")
+        errors =  run_mrf_insert(tiles, mrf, insert_method, resize_resampling, target_x, target_y, mrf_blocksize,
+                             target_extents, target_epsg, nodata, merge, working_dir)
+    else: 
+        log_info_mssg("making parallel call with length of tiles is {}, mrf is {}\n".format(len(tiles), mrf))
+
+        func = functools.partial(run_mrf_insert, mrf=mrf, insert_method = insert_method, \
+            resize_resampling = resize_resampling, target_x = target_x, target_y = target_y, \
+                mrf_blocksize = mrf_blocksize, target_extents = target_extents, target_epsg = target_epsg, \
+                    nodata = nodata, merge = merge, working_dir = working_dir, parallel=True)
+
+        with open(mrf) as f: # make mp_safe
+            data = f.read()
+            data = data.replace("<Raster>", "<Raster mp_safe=\"on\">")
+        
+        with open(mrf, "w") as f: # overwrite mrf
+            f.write(data)
+
+        log_info_mssg("Splitting list of length {} into chunks of size {}".format(len(tiles), len(tiles) // no_pools))
+
+        def chunks(l, n):
+            for i in xrange(0, len(l), n):
+                yield l[i:i+n]
+
+        partition = chunks(tiles, len(tiles) // no_pools)
+
+        with poolcontext(processes=no_pools) as pool:
+            results = pool.map(func, partition, 1)
+
+        log_info_mssg("mrf {} map finished, results are type {}, {}".format(mrf, type(results), results))
+
+        errors = sum(results)
+
+    log_info_mssg("Errors {}, mrf {}".format(errors, mrf))
+
+    return errors
+
+def run_mrf_insert(tiles, mrf, insert_method, resize_resampling, target_x, target_y, mrf_blocksize,
```

Comment:
If there is a separate `parallel_mrf_insert` function, what purpose does having a new `parallel` argument do here?